processing/src/main/java/io/druid/query/extraction/StringFormatExtractionFn.java

```diff
@@ -33,14 +33,26 @@
 public class StringFormatExtractionFn extends DimExtractionFn
 {
   private final String format;
+  private final boolean nullAsNull;
+  private final boolean nullAsEmpty;
 
   @JsonCreator
   public StringFormatExtractionFn(
-      @JsonProperty("format") String format
+      @JsonProperty("format") String format,
+      @JsonProperty("nullAsNull") boolean nullAsNull,
+      @JsonProperty("nullAsEmpty") boolean nullAsEmpty
   )
   {
     Preconditions.checkArgument(!Strings.isNullOrEmpty(format), "format string should not be empty");
+    Preconditions.checkArgument(!(nullAsNull && nullAsEmpty), "nullAsNull and nullAsEmpty cannot be both true");
     this.format = format;
+    this.nullAsNull = nullAsNull;
+    this.nullAsEmpty = nullAsEmpty;
```

Comment:
Would that be better handled with a 3-value enum for "null handling behavior", vs. using booleans?